source/postprocess/plateness_statistics.cc

```diff
@@ -0,0 +1,172 @@
+                    {
+                      // If the cell is at the top boundary, assign face_idx.
+                      face_idx = f;
+                      at_upper_surface = true;
+                      break;
+                    }
+                }
+
+            }
+            if (at_upper_surface)
+              {
+                fe_face_values.reinit(cell, face_idx);
+                fe_face_values[this->introspection().extractors.velocities].get_function_symmetric_gradients (this->get_solution(), strain_rate);
+              
+                for (unsigned int q = 0; q<fe_face_values.n_quadrature_points; ++q)
+                  {
+                    local_second_invariant_of_strain_rate_and_corresponding_area[q].first = std::max(std::sqrt(std::fabs(second_invariant(deviator(strain_rate[q])))),min_strain_rate);
+                    local_second_invariant_of_strain_rate_and_corresponding_area[q].second = fe_face_values.JxW(q);
+           
+                    // second invariant of strain rate over the whole surface
+                    local_second_invariant_of_strain_rate_integral += local_second_invariant_of_strain_rate_and_corresponding_area[q].first * local_second_invariant_of_strain_rate_and_corresponding_area[q].second;
+                    // surface area over the whole surface
+                    local_surface_area_integral += local_second_invariant_of_strain_rate_and_corresponding_area[q].second;
+                  }
+              }
+          }
+      const double total_second_invariant_of_strain_rate = Utilities::MPI::sum (local_second_invariant_of_strain_rate_integral, this->get_mpi_communicator());
+      const double total_surface_area = Utilities::MPI::sum (local_surface_area_integral, this->get_mpi_communicator());
+
+      //sort local second invariants of strain rate in size order
+      std::sort(local_second_invariant_of_strain_rate_and_corresponding_area.begin(), local_second_invariant_of_strain_rate_and_corresponding_area.end());
+
+      //find the fraction of the surface area where 80% of the deformation occurs
+      //double f_80 = 0.0;
+      double cumulative_second_invariant_of_strain_rate = 0.0;
+      double cumulative_surface_area = 0.0;
+      int row_number = 0;
+      do
+        {
+          Assert(row_number < local_second_invariant_of_strain_rate_and_area.size(), ExcInternalError());
+          cumulative_second_invariant_of_strain_rate = cumulative_second_invariant_of_strain_rate + local_second_invariant_of_strain_rate_and_corresponding_area[row_number].first * local_second_invariant_of_strain_rate_and_corresponding_area[row_number].second;
+          cumulative_surface_area = cumulative_surface_area + local_second_invariant_of_strain_rate_and_corresponding_area[row_number].second;
+          row_number = row_number + 1;
+        }
+      while (cumulative_second_invariant_of_strain_rate<0.8*total_second_invariant_of_strain_rate);
```

Comment:
As a consequence of the comment above, `local_second_invariant_of_strain_rate_and_corresponding_area` has length zero, and so the assertion in line 108 will fail in the first go-around already. Based on how you index the array via ` local_second_invariant_of_strain_rate_and_corresponding_area[q]`, one could think that you really wanted the array to have size `n_q_points`, but that's not going to work either: I *think* what you need is an array with as many rows as there are cells, so that you can select a number of cells that together make up a certain fraction of the total plateness.